### PR TITLE
Re-enable terraform orb collapsible plans

### PR DESCRIPTION
**Pull request by chris-brindley**
This PR fixes the Terraform and Terraform-v2 orbs PR comment matching functionality. The comment matching method wasn't updated when the format of the comment was updated, so the orb became unable to correctly detect plans / update comments on PRs, etc.

It also adds a stage to the CI pipelines for these two orbs to run the existing python tests, so that this problem won't crop up again in the future (hopefully)

### CODE REVIEW

**Comment by bnjns** [comment]
Hey @chris-brindley; it appears both of us have started looking into this 😂 I've got a PR open to revert the version which introduced this break (#355) before I took a look at the issue, but would you rather just continue with this PR? Conscious we're getting a few teams noticing this issue and its blocking their pipelines.

**Comment by chris-brindley** [comment]
Happy for you to revert for now to unblock teams. I can continue working on this and add the functionality back in once I'm happy with it :)

**Comment by bnjns** [comment]
Awesome, thanks! Happy to review if you want 🙂

**Comment by chris-brindley** [comment]
Just working on testing it in a real live pipeline, I'll ping you for a review once I'm comfortable the issue is resolved

**Review by bnjns — COMMENTED** [review]
Thanks very much for taking a look at this, and adding the tests to CI! Just a couple of comments/questions 🙂

File: terraform-v2/comment_util.py
Comment (chris-brindley) [reply terraform-v2/comment_util.py]: Agreed that maybe we don't need to support all styles for all time, but I think maybe we can support just the current style and the previous style - that way we are minimising the potential for disruption whilst keeping it maintainable.

File: terraform-v2/comment_util.py
Comment (chris-brindley) [reply terraform-v2/comment_util.py]: Nice idea, I've implemented this in https://github.com/ovotech/circleci-orbs/pull/354/commits/78a0cba89f7d372928498fab00221cebe34ffe98

File: terraform-v2/comment_util.py
Comment (bnjns) [reply terraform-v2/comment_util.py]: Yeah, I think what you've done with the list is a pretty good solution

**Comment by chris-brindley** [comment]
I've tested these updates on a live pipeline and it seems to be working well.

**Review by bnjns — APPROVED** [review]
Awesome stuff, thanks 🎉 

Might be good to get a second pair of eyes from someone else, just to double check?

**Review by eversC — APPROVED** [review]
lgtm, thanks Ben + Chris!

**Pull request merged**
